app/src/main/java/com/eveningoutpost/dexdrip/G5Model/Ob1G5StateMachine.java

```diff
@@ -627,10 +614,21 @@ public static boolean doGetData(Ob1G5CollectionService parent, RxBleConnection c
                                 DexSessionKeeper.clearStart();
                             }
                             if (Pref.getBooleanDefaultFalse("ob1_g5_preemptive_restart")) {
-                                if (txtime.getSessionDuration() > Constants.DAY_IN_MS * (usingG6() ? 9 : 6)
+                                int restartDaysThreshold = usingG6() ? 9 : 6;
+                                if (txtime.getSessionDuration() > Constants.DAY_IN_MS * restartDaysThreshold
                                         && txtime.getSessionDuration() < Constants.MONTH_IN_MS) {
-                                    UserError.Log.uel(TAG, "Requesting preemptive session restart");
-                                    restartSensorWithTimeTravel();
+                                    BestGlucose.DisplayGlucose displayGlucose = BestGlucose.getDisplayGlucose();
+                                    // Defer restart up to 12h if current delta is high, which can lead to sensor
+                                    // errors if session is restarted during times of high fluctuation
+                                    if (displayGlucose != null && Math.abs(displayGlucose.delta_mgdl) <= 4
```

Comment:
This potential to shift the restart time by 12 hours is something I am nervous about as people have been told the restart will be at the same time as the original start.  I would like to have an additional preference option in the G5/G6 debugging options to control this and just for now to default this to `false` but with a plan to make it default to `true` in the near future. That way nightly users who update will still get the old behavior even once we change the default but new users will get the new behavior.

I also want to do the same for the longer time-travel as we know the 2 hours works but this is potentially a bit more risky. So we would want a new option for that as well to say that the preemptive restarts can adjust the time to minimize reading jumps.